kedro/extras/datasets/spark/spark_hive_dataset.py

```diff
@@ -151,6 +158,13 @@ def __init__(
         self._database = database
         self._table = table
         self._stage_table = "_temp_" + table
+        self._partition = partition
+
+        # get the name of each partition
+        self._partitions: List[str] = []
+        if self._partition is not None:
+            for partition_set in self._partition.split(","):
+                self._partitions.append(partition_set.split("=")[0].strip())
```

Comment:
Would it make sense to just directly make the function argument `partition: List[str] = None`, as is done for `table_pk`? This would do away with the need for string manipulation here.

Reply:
That makes sense, but with just the list of column names we can't identify a partition. Maybe we can make `partition` get a list of tuples, where each position of the list identifies the column name and value. What do you think?

Reply:
Ah I see. That could indeed work, but now I'm wondering exactly what the `partition` string in the insert could look like. e.g. we can have `partition_1 = x, partition_2 = y` but we could also just have `partition_1, partition_2` right? Do you know what is the full specification of the syntax of this string?

Now I'm actually also wondering why `SparkHiveDataSet` is using dynamically build SQL queries in the first place rather than the Python API, which would presumably make this sort of thing much easier. Is that impossible when doing the hive thing? @jiriklein

Reply:
Hi @AntonyMilneQB, sorry for the late reply.
That's right, the `partition` string will be something like `partition_1 = x, partition_2 = y` if the partition type is static or `partition_1, partition_2` if the partition is dynamic.